manim/utils/polylabel.py

```diff
@@ -0,0 +1,152 @@
+        """
+        Initializes the Polygon with the given rings.
+
+        Parameters
+        ----------
+        rings : list[np.ndarray]
+            List of arrays, each representing a polygonal ring
+        """
+        # Flatten Array
+        csum = np.cumsum([ring.shape[0] for ring in rings])
+        self.array = np.concatenate(rings, axis=0)
+
+        # Compute Boundary
+        self.start = np.delete(self.array, csum - 1, axis=0)
+        self.stop = np.delete(self.array, csum % csum[-1], axis=0)
+        self.diff = np.delete(np.diff(self.array, axis=0), csum[:-1] - 1, axis=0)
+        self.norm = self.diff / np.einsum("ij,ij->i", self.diff, self.diff).reshape(
+            -1, 1
+        )
+
+        # Compute Centroid
+        x, y = self.start[:, 0], self.start[:, 1]
+        xr, yr = self.stop[:, 0], self.stop[:, 1]
+        self.area = 0.5 * (np.dot(x, yr) - np.dot(xr, y))
+        if self.area:
+            factor = x * yr - xr * y
+            cx = np.sum((x + xr) * factor) / (6.0 * self.area)
+            cy = np.sum((y + yr) * factor) / (6.0 * self.area)
+            self.centroid = np.array([cx, cy])
+
+    def compute_distance(self, point: np.ndarray) -> float:
+        """Compute the minimum distance from a point to the polygon."""
+        scalars = np.einsum("ij,ij->i", self.norm, point - self.start)
+        clips = np.clip(scalars, 0, 1).reshape(-1, 1)
+        d = np.min(np.linalg.norm(self.start + self.diff * clips - point, axis=1))
+        return d if self.inside(point) else -d
+
+    def inside(self, point: np.ndarray) -> bool:
+        """Check if a point is inside the polygon."""
+        # Views
+        px, py = point
+        x, y = self.start[:, 0], self.start[:, 1]
+        xr, yr = self.stop[:, 0], self.stop[:, 1]
+
+        # Count Crossings (enforce short-circuit)
+        c = (y > py) != (yr > py)
+        c = px < x[c] + (py - y[c]) * (xr[c] - x[c]) / (yr[c] - y[c])
+        return np.sum(c) % 2 == 1
+
+
+class Cell:
+    def __init__(self, c: np.ndarray, h: float, polygon: Polygon) -> None:
+        """
+        Initializes the Cell, a square in the mesh covering the polygon.
+
+        Parameters
+        ----------
+        c : np.ndarray
+            Center coordinates of the Cell.
+        h : float
+            Half-Size of the Cell.
+        polygon : Polygon
+            Polygon object for which the distance is computed.
+        """
+        self.c = c
+        self.h = h
+        self.d = polygon.compute_distance(self.c)
+        self.p = self.d + self.h * np.sqrt(2)
+
+    def __lt__(self, other: Cell) -> bool:
+        return self.d < other.d
+
+    def __gt__(self, other: Cell) -> bool:
+        return self.d > other.d
+
+    def __le__(self, other: Cell) -> bool:
+        return self.d <= other.d
+
+    def __ge__(self, other: Cell) -> bool:
+        return self.d >= other.d
+
+
+def PolyLabel(rings: list[list[Point3D]], precision: float = 0.01) -> Cell:
```

Comment:
`PascalCase` is reserved for class definitions. For functions, use `snake_case` instead.

`rings` could be a `Sequence[Point3D_Array]` instead.
```suggestion
def polylabel(rings: Sequence[Point3D_Array], precision: float = 0.01) -> Cell:
```